### PR TITLE
Revert and fix upgrade to govuk frontend v3

### PR DESCRIPTION
**Pull request by injms**
## What
Reverts the [revert](https://github.com/alphagov/service-manual-frontend/pull/539) of [the upgrade](https://github.com/alphagov/service-manual-frontend/pull/518) that broke service toolkit page and fixes the layout issues that were present on the Service Toolkit page.

## Why
The Service Toolkit page still used Frontend Toolkit for layout, so looked broken.

## Quick checks 

* https://govuk-service-manual-fr-pr-542.herokuapp.com/service-toolkit
* https://govuk-service-manual-fr-pr-542.herokuapp.com/service-manual/
* https://govuk-service-manual-fr-pr-542.herokuapp.com/service-manual/helping-people-to-use-your-service
* https://govuk-service-manual-fr-pr-542.herokuapp.com/service-manual/design
* https://govuk-service-manual-fr-pr-542.herokuapp.com/service-manual/service-assessments
* https://govuk-service-manual-fr-pr-542.herokuapp.com/service-manual/service-standard
* https://govuk-service-manual-fr-pr-542.herokuapp.com/service-manual/service-standard/point-1-understand-user-needs
* https://govuk-service-manual-fr-pr-542.herokuapp.com/service-manual/communities
* https://govuk-service-manual-fr-pr-542.herokuapp.com/service-manual/communities/accessibility-community

### CODE REVIEW

**Comment by 36degrees** [comment]
Really minor thing, and it could be fixed in a future pull request, but the show/hide all updates links do not use the new focus style.

Screen Shot 2019-09-18 at 15 36 05

Otherwise, this is looking good to me.

**Comment by 36degrees** [comment]
In the same category of nice-to-fix-but-not-essential, the 'is there anything wrong with this page' form seems to have a double gutter on mobile:

Screen Shot 2019-09-18 at 15 37 51

**Comment by injms** [comment]
> I'm happy that the code was previously reviewed as part of the original PR (before it was reverted), so I've not reviewed the code – just looked at the example pages linked from this PR.

There are commits since the revert that weren't part of the code review - these fix the problem that caused the reversion:

* [0e01ec8](https://github.com/alphagov/service-manual-frontend/pull/542/commits/0e01ec835a88d10d3732c856b5d36faeb50c09bd)
* [b6d1be9](https://github.com/alphagov/service-manual-frontend/pull/542/commits/b6d1be9b9b7f560d106d6bbcb96c669809e31990)
* [ea38440](https://github.com/alphagov/service-manual-frontend/pull/542/commits/ea3844087d191e6262dfd5efd63aec3a2136b84e)

**Comment by vanitabarrett** [comment]
The spacing between paragraphs and headings seems a bit small compared to how it used to be? It also doesn't quite match what the govspeak component should be doing (p tags should have 20px top and bottom margin, these seem to have 15px), so i'm wondering if something is interfering?
Screen Shot 2019-09-19 at 10 17 41

I can also see some CSS errors but I can't work out where they're coming from.
Screen Shot 2019-09-19 at 10 18 53

This is on https://govuk-service-manual-fr-pr-542.herokuapp.com/service-manual/service-standard/point-1-understand-user-needs

**Comment by 36degrees** [comment]
> There are commits since the revert that weren't part of the code review

Aside from @vanitabarrett's comments, those commits make sense to me 👍

**Comment by chao-xian** [comment]
Thanks @vanitabarrett @36degrees

File: app/views/content_items/service_toolkit.html.erb [hunk 1]
```diff
@@ -1,38 +1,43 @@
...
+  <div class="app-c-hero__content govuk-width-container">
+    <div class="govuk-main-wrapper">
+      <div class="govuk-grid-row">
+        <div class="govuk-grid-column-two-thirds">
+          <h1 class="app-c-hero__title govuk-heading-xl govuk-!-margin-bottom-6 app-c-hero__heading--inverse">Design and build government services</h1>
 
-        <p class="hero__description">
-          All you need to design, build and run services that meet government standards.
-        </p>
+          <p class="app-c-hero__descriptio govuk-body-l govuk-!-margin-bottom-3 app-c-hero__body--inverse">
```
[the remark this answers is not in the document]
Comment (injms) [reply app/views/content_items/service_toolkit.html.erb]: Changed to remove component `-c-` from the classes.

File: app/views/content_items/service_toolkit.html.erb [hunk 2]
```diff
@@ -1,38 +1,43 @@
...
   content_for :header_title, "Service Toolkit"
-  content_for :body_classes, "full-width"
+  content_for :body_classes, "app-full-width"
 %>
-<header class="hero">
-  <div class="hero__content">
-    <div class="grid-row">
-      <div class="column-two-thirds">
-        <h1 class="hero__title">Design and build government services</h1>
+<header class="app-c-hero">
```
[the remark this answers is not in the document]
Comment (injms) [reply app/views/content_items/service_toolkit.html.erb]: Changed to remove component `-c-` from the classes.

**Comment by injms** [comment]
Good catch on the CSS errors @vanitabarrett!

The CSS errors caused by `em(4)` come from a [GOV.UK Elements](https://github.com/alphagov/govuk_elements/blob/master/assets/sass/elements/_elements-typography.scss#L142), which is being included via Frontend Toolkit. There is a [helper function in GOV.UK Elements](https://github.com/alphagov/govuk_elements/blob/master/assets/sass/elements/_helpers.scss#L6) that _should_ output valid CSS, but it doesn't appear to be working. Checking against the live site, the errors made it into production 😳.

Since Elements and Frontend Toolkit are deprecated, are due to be removed soon 🤞, and this error will be ignored by browsers, I think that this can be safely ignored.

**Comment by injms** [comment]
The paragraph tags are set to have a lower spacing of 15px because ~`¯\_(ツ)_/¯`~ that's what the CSS in this app originally had. I missed this thanks to the loveliness of the way margins collapse.

I originally added an increase in line height so the focus state didn't overlap on the next line. Since it's just-about-okay, I'll add that as an issue for discussion on the component itself.

Anyways, this is a longwinded way of me saying I've removed the app specific Govspeak overrides, so the spacing should be more consistent.

**Pull request merged**